docs/api.md

```diff
@@ -30,6 +30,12 @@ power state on the physical host.
 host. For example, a Machine when the host is being used by the
 machine-api.
 
+*externallyProvisioned* -- A boolean indicating whether something else
+is managing the image running on the host. When set, if no image is
+provided, the host's power status and hardware inventory will be
+monitored. If the *image* contains image parameters, this field is
+ignored.
```

Comment:
why not just always ignore the image field if externallyProvisioned is true?

Reply:
The user can change the externallyProvisioned field at any time. So, they could start to provision a host in the normal way and then while it's in the middle of provisioning set externallyProvisioned=true. At that point, the logic managing the provisioning workflow would be circumvented and the host status would never update properly.

We _could_ allow that, but it's easier to reason about the host if we say that giving an image and saying a host is externally provisioned are mutually exclusive, but the image has precedence if both are provided. We end up with fewer overall transitions in the state diagram that way.

Reply:
OK, that makes sense, thanks.

Reply:
I added some comments in the code to explain the reasoning for this choice.